x-pack/test/apm_api_integration/tests/service_overview/instance_details.spec.ts

```diff
@@ -46,53 +47,93 @@ export default function ApiTest({ getService }: FtrProviderContext) {
 
-          response = await apmApiClient.readUser({
-            endpoint:
-              'GET /internal/apm/services/{serviceName}/service_overview_instances/details/{serviceNodeName}',
-            params: {
-              path: { serviceName: 'opbeans-java', serviceNodeName: serviceNodeIds[0] },
-              query: {
-                start,
-                end,
-              },
-            },
-          });
-        });
+    const serviceInstance = apm
+      .service({ name: 'service1', environment: 'production', agentName: 'go' })
+      .instance('multiple-env-service-production');
+
+    const metricOnlyInstance = apm
+      .service({ name: 'service1', environment: 'production', agentName: 'java' })
+      .instance('metric-only-production');
+
+    before(async () => {
+      return synthtrace.index([
+        range
+          .interval('1s')
+          .rate(4)
+          .generator((timestamp) =>
+            serviceInstance
+              .transaction({ transactionName: 'GET /api' })
+              .timestamp(timestamp)
+              .duration(1000)
+              .success()
+          ),
+        range
+          .interval('30s')
+          .rate(1)
+          .generator((timestamp) =>
+            metricOnlyInstance
+              .containerId('123')
+              .podId('234')
+              .appMetrics({
+                'system.memory.actual.free': 1,
+                'system.cpu.total.norm.pct': 1,
+                'system.memory.total': 1,
+                'system.process.cpu.total.norm.pct': 1,
+              })
+              .timestamp(timestamp)
+          ),
+      ]);
+    });
+
+    after(() => {
+      return synthtrace.clean();
+    });
+
+    describe('fetch instance details', () => {
+      let response: {
+        status: number;
+        body: ServiceOverviewInstanceDetails;
+      };
 
-        it('returns the instance details', () => {
-          expect(response.body).to.not.eql({});
+      let serviceNodeIds: string[];
+
+      before(async () => {
+        serviceNodeIds = await getServiceNodeIds({
+          apmApiClient,
+          start,
+          end,
+          serviceName: 'service1',
         });
 
-        it('return the correct data', () => {
-          expectSnapshot(omit(response.body, '@timestamp')).toMatch();
+        response = await apmApiClient.readUser({
+          endpoint:
+            'GET /internal/apm/services/{serviceName}/service_overview_instances/details/{serviceNodeName}',
+          params: {
+            path: { serviceName: 'service1', serviceNodeName: serviceNodeIds[0] },
+            query: {
+              start,
+              end,
+            },
+          },
         });
       });
-    }
-  );
+
+      it('returns the instance details', () => {
+        expect(response.body).to.not.eql({});
+      });
+
+      it('return the correct data', () => {
+        expectSnapshot(omit(response.body, '@timestamp')).toMatch();
```

Comment:
Do we need to keep the snapshot test?

Reply:
Yes, we have removed the archives, snapshots are generated based on the available data

Reply:
Fine to leave the snapshots. But we should be careful about snapshot testing. It's very easy to make a breaking change to the implementation and then blindly replace the snapshot. That's much harder to do when expecting specific (hardcoded) values.